### PR TITLE
Improve yast2log error parsing

### PR DESCRIPTION
**Pull request by rwx788**
We aim 2 things with this change:
* detect new errors in the logs
* make relevant logs output visible on the job page in order to simplify
investigation (no need to download logs)

See [poo#48137](https://progress.opensuse.org/issues/48137).

## Verification runs
* [sle 15](http://f174.suse.de/tests/150)
* [sle 12](http://f174.suse.de/tests/151) (Scenario was not enabled for SLE 12, I've added it to the YaST job group)
* [sle 15 new generic error](http://f174.suse.de/tests/144#step/logs_from_installation_system/372)
* [sle 15 specific error detected](http://f174.suse.de/tests/153#)

### CODE REVIEW

**Comment by okurz** [comment]
LGTM

Code and also the two verification runs look fine however do you have a failing test as example so that we can see we actually have some text from an "internal error" that we could copy into a bug?

**Comment by rwx788** [comment]
> LGTM
> 
> Code and also the two verification runs look fine however do you have a failing test as example so that we can see we actually have some text from an "internal error" that we could copy into a bug?

Let me remove one of the known errors, plus here is the run where there is new generic error:
http://f174.suse.de/tests/144#step/logs_from_installation_system/372

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -143,48 +143,146 @@ sub investigate_yast2_failure {
...
+        "No textdomain configured"                   => 'bsc#1127756',    # Detecting missing translations
+                                                                          # Detecting specifi errors proposed by the YaST dev team
+        "nothing provides"                           => undef,            # Detecting missing required packages
+        "but this requirement cannot be provided"    => undef,            # Detecting package conflicts
+        "Could not load icon|Couldn't load pixmap"   => undef,            # Detecting missing icons
+        "Internal error. Please report a bug report" => undef,            # Detecting internal errors
+
+        # The error below will be cleaned up, see https://trello.com/c/5qTQZKH3/2918-sp2-logs-cleanup
+        # Adding reference to trello
+        "<3>.*Error output: dracut:"                            => 'https://trello.com/c/5qTQZKH3/2918-sp2-logs-cleanup',
```
Comment (asmorodskyi) [inline lib/opensusebasetest.pm]:
```
sub yast_trello() {
 my $card= shift;
return 'https://trello.com/c/5qTQZKH3/' . $card;
}
```

```suggestion
        "<3>.*Error output: dracut:"                            => yast_trello('2918-sp2-logs-cleanup');
```

Comment (rwx788) [reply lib/opensusebasetest.pm]: I'm sorry, but that's overkill. I can put that string in the variable if you like it more.

Comment (asmorodskyi) [reply lib/opensusebasetest.pm]: just an idea , you may ignore it if you don't like it. the best would have something like for bugs , simply writing `trl#2918-sp2-logs-cleanup` but I am not sure how it is done for bugzilla so can't suggest something certain. It just hurt my eye when I see copy/paste of same string so many times but **you** are main reader of this code so again it is up to you :)

**Comment by asmorodskyi** [comment]
LGTM

File: lib/opensusebasetest.pm (hunk 1 above)
Comment (rwx788) [reply lib/opensusebasetest.pm]: For trello the board can be different, and actually just `https://trello.com/c/5qTQZKH3/` points to the same trello card, but board url is `https://trello.com/b/aICWq7sT/agile-yast-incoming-board` So there is not easy way. Having so many copy pasted lines is mainly to accept current state of the things, which isn't that clean as we can see =)

**Pull request merged**